src/crimson/os/seastore/segment_manager/zns.cc

```diff
@@ -101,6 +96,24 @@ struct ZoneReport {
   }
 };
 
+static seastar::future<> get_blk_dev_size(
+  seastar::file &device,
+  uint64_t &size)
```

Comment:
`static seastar::future<std::size_t> get_blk_dev_size(seastar::file &device)`

Reply:
> There is a compile error, see make check.

I think this is because WITH_ZBD is used in all make files except the below one
https://github.com/ceph/ceph/blob/12aade9b2c2aed257c2bb72cf839ea3d258e6f5b/src/crimson/os/seastore/CMakeLists.txt

Here WITH_ZNS is being used. I am working on this now and will fix it.

Reply:
That is a different thread.

What I meant here is that `size` is more about return value, not a parameter. The signature `static seastar::future<std::size_t> get_blk_dev_size(seastar::file &device)` is more reasonable here.

Reply:
>That is a different thread.
Sorry, that comment was for the make_check failing issue. I have added another commit to this PR to fix it.

I have updated `static seastar::future<> get_blk_dev_size(seastar::file &device, uint64_t &size)` to 
 `static seastar::future<std::size_t> get_blk_dev_size(seastar::file &device)`